source/common/upstream/cluster_manager_impl.cc

```diff
@@ -617,10 +617,17 @@ bool ClusterManagerImpl::addOrUpdateCluster(const envoy::config::cluster::v3::Cl
   const auto existing_active_cluster = active_clusters_.find(cluster_name);
   const auto existing_warming_cluster = warming_clusters_.find(cluster_name);
   const uint64_t new_hash = MessageUtil::hash(cluster);
-  if ((existing_active_cluster != active_clusters_.end() &&
-       existing_active_cluster->second->blockUpdate(new_hash)) ||
-      (existing_warming_cluster != warming_clusters_.end() &&
-       existing_warming_cluster->second->blockUpdate(new_hash))) {
+  if (existing_warming_cluster != warming_clusters_.end()) {
```

Comment:
@mattklein123 @tbarrella  do you think the same issue may exist in LDS in https://github.com/envoyproxy/envoy/blob/3bddd1a8f8f67174d487a06c690f67b944b9e0da/source/server/listener_manager_impl.cc#L397?

It's hard for me to tell, since the logic here is subtle and orders are different.

Reply:
That's from #12645 which looks like it was meant to fix the same issue with LDS. I guess the approach is different; I didn't do it that way because it seemed simpler to take the newest config as a typical update rather than introduce cancellation

Reply:
Would it make sense to factor out this pattern? Divergence seems scary :)

Reply:
I hear what you mean in terms of divergence. For this do you mean actual code refactoring (I'm not sure about the ROI here) or just making sure the logic is the same? What are your thoughts on the LDS (cancel the warming listener and block the current update) vs. CDS approach (accept the current update)?

Reply:
I don't feel strongly on how we do it as long as we make them consistent and keep them that way (code comments explaining when to update what? Structural is always nicer if it's not too much complexity).

I think whatever is simplest here makes sense, but CC @adisuissa @dmitri-d